### PR TITLE
Disable Implicit casts

### PR DESCRIPTION
**Pull request by klavs**
add analyzer rule to avoid analyzable errors

### CODE REVIEW

**Comment by comigor** [comment]
Hey @klavs, thanks for those PRs!
Checks seem to be broken. I'll try to take a look at it later today and merge all those PRs (and display linter error on GitHub actions output)

File: lib/generator.dart [hunk 1]
```diff
@@ -81,11 +89,7 @@ QueryDefinition generateQuery(
   final List<Definition> inputsClasses = [];
   if (operation.variableDefinitions != null) {
     operation.variableDefinitions.forEach((v) {
-      NamedTypeNode unwrappedType = v.type;
-
-      if (v.type is ListTypeNode) {
-        unwrappedType = (v.type as ListTypeNode).type;
-      }
+      NamedTypeNode unwrappedType = _unwrapType(v.type);
```
[the remark this answers is not in the document]
Comment (amadeu01) [reply lib/generator.dart]: I forgot to send the review, but that comment was before seeing the PR about that.

**Comment by amadeu01** [comment]
@klavs  Although analyzable errors could be avoided easily, I guess it would be nice some test that would break before that change (I was trying to elaborate one test for this, but I couldn't, I'll try more at home)

**Comment by amadeu01** [comment]
@klavs just opened a PR with the test I was expecting.

**Comment by comigor** [comment]
Released as https://pub.dev/packages/artemis/versions/1.0.3

**Comment by comigor** [comment]
I've also updated GH action to display linter errors on failure

**Pull request merged**